src/transaction/flashback_cl.c

```diff
@@ -0,0 +1,197 @@
+ *
+ * return              : error code
+ * ptr (in/out)        : memory pointer where to unpack
+ * summary (out)       : brief summary information
+ * num_summary (in)    : number of summary to unpack
+ * classname_list (in) : classnames
+ * oidlist (in)        : OID of classnames
+ */
+
+int
+flashback_unpack_and_print_summary (char **ptr, FLASHBACK_SUMMARY_INFO_MAP * summary, int num_summary,
+				    dynamic_array * classname_list, OID * oidlist)
+{
+  TRANID trid;
+  char *user = NULL;
+  time_t start_time, end_time;
+  int num_insert, num_update, num_delete;
+  LOG_LSA start_lsa, end_lsa;
+  int num_table;
+  OID classoid;
+  char *tmp_ptr = *ptr;
+
+  bool stop_print = false;
+  int max_window_size = flashback_util_get_winsize ();
+  int len = 0;
+
+  printf
+    ("  transaction id  |    user   |   start time    |   end time   |   num_insert   |   num_update   |   num_delete   |   tables   ");
+  len++;
+
+  for (int i = 0; i < num_summary; i++)
+    {
+      FLASHBACK_SUMMARY_INFO info = { 0, LSA_INITIALIZER, LSA_INITIALIZER };
+
+      /* testing code, it will be replaced with print function */
+      tmp_ptr = or_unpack_int (tmp_ptr, &trid);
+      tmp_ptr = or_unpack_string_nocopy (tmp_ptr, &user);
+      tmp_ptr = or_unpack_int64 (tmp_ptr, &start_time);
+      tmp_ptr = or_unpack_int64 (tmp_ptr, &end_time);
+      tmp_ptr = or_unpack_int (tmp_ptr, &num_insert);
+      tmp_ptr = or_unpack_int (tmp_ptr, &num_update);
+      tmp_ptr = or_unpack_int (tmp_ptr, &num_delete);
+      tmp_ptr = or_unpack_log_lsa (tmp_ptr, &start_lsa);
+      tmp_ptr = or_unpack_log_lsa (tmp_ptr, &end_lsa);
+      tmp_ptr = or_unpack_int (tmp_ptr, &num_table);
+
+      info.trid = trid;
+      info.start_lsa = start_lsa;
+      info.end_lsa = end_lsa;
+
+      /* *INDENT-OFF* */
+      summary->emplace (trid, info);
+      /* *INDENT-ON* */
+
+      if (!stop_print)
+	{
+	  char stime_buf[20];
+	  char etime_buf[20];
+
+	  strftime (stime_buf, 20, "%d-%m-%Y:%H%M%S", localtime (&start_time));
+	  strftime (etime_buf, 20, "%d-%m-%Y:%H%M%S", localtime (&end_time));
+
+	  printf ("\t%d\t%s\t%s\t%d\t%d\t%d\t\t", trid, user, stime_buf, etime_buf, num_insert, num_update, num_delete);
+	}
+
+      for (int j = 0; j < num_table; j++)
+	{
+	  int idx = 0;
+	  char classname[SM_MAX_IDENTIFIER_LENGTH + 1] = "\0";
+
+	  tmp_ptr = or_unpack_oid (tmp_ptr, &classoid);
+	  if (!stop_print)
+	    {
+	      idx = flashback_find_class_index (classname_list, oidlist, classoid);
+	      if (idx == -1)
+		{
+		  /* er_set */
+		  return ER_FLASHBACK_INVALID_CLASS;
+		}
+
+	      da_get (classname_list, idx, classname);
+	      printf ("%s ", classname);
+	    }
+	}
+
+      if (!stop_print)
+	{
+	  printf ("\n");
+	  len++;
+	}
+
+      if (len >= max_window_size)
+	{
+	  char c;
+	  printf ("press 'q' to quit or press anything to continue");
+	  c = getchar ();
+	  if (c == 'q')
+	    {
+	      stop_print = true;
+	    }
+
+	  len = 0;
+	  printf ("\033[A\33[2K\r");
```

Comment:
What does it mean?

Reply:
It means, erase the line `"press 'q' to quit or press anything to continue"`.
because above message is no more required when user press any key.

Reply:
Aha! I've got it. How about leaving a comment?

Reply:
Oh, It seems better! I will leave a comment

Reply:
Is it work on Windows too?

And when the user chooses the 'q', I think that it is enough just to finish the output of the summary info and return success in the function. Is there any reason to use the stop_print flag and to keep going the execution? I think that you don't gather rest summary info because it is determined that the user already found the necessary information(transaction id).

Reply:
@hornetmj 
1. I have to test on Windows.
2. The case you mentioned is when the user uses it normally. 
    This is in case the user executes it more than once. 
    So user knows that there is a transaction after the printed summary information so far.